wfapi/error.go

```diff
@@ -112,39 +110,38 @@ func ErrorWorkspace(wsPath string, cause error) error {
 //
 //    - warpforge-error-executor-failed --
 func ErrorExecutorFailed(executorEngineName string, cause error) error {
-	result := serum.Errorf(ECodeExecutorFailed,
-		"executor engine failed: the %s engine reported error: %w", executorEngineName, cause)
-	addDetails(result, [][2]string{
-		{"engineName", executorEngineName},
+	return serum.Error(ECodeExecutorFailed, serum.WithCause(cause),
+		serum.WithMessageTemplate("the {{engineName|q}} engine reported error"),
+		serum.WithDetail("engineName", executorEngineName),
 		// ideally we'd have more details here, but honestly, our executors don't give us much clarity most of the time, so... we'll see.
-	})
-	return result
+	)
 }
 
+// DEPRECATED: This is just adds a degenerate repetition of the error code
```

Comment:
Not sure if agree on this; it also nudges to attach a path.

(I'm not sure if all IO errors have this already and it's redundant; I think some don't?  I guess the ideal would be to only bother to attach the path on those types where it's not redundant, but this function just didn't grow to be that clever yet.)

Reply:
If we template the path into the message then I'd be more inclined to agree. Untemplated details only show up during serialization and not in the message user's see (unless you are in JSON mode). And some io errors not having paths makes it seem like we should have something more specific for io errors with paths. As it stands this just seems out-of-place and we should just call serum directly. We're mostly being saved by the fact that most io errors being wrapped happen to emit file paths.